### PR TITLE
node-pruning

### PR DESCRIPTION
**Pull request by dblane-digicatapult**
@PierreBesson Could you take a look at this and see if I'm on the right track.

#34 Doesn't mention exactly what options you are looking for, I've run into some issues when using `--validator` and `--pruning <value>` even with `--unsafe-pruning`

### CODE REVIEW

**Comment by PierreBesson** [comment]
What I would do is always add `--unsafe-pruning` when pruning is enabled (ie defined or `>0`). As defined in `polkadot --help`, pruning is either a number or the `archive` string. If pruning equals 0, we could set the `archive` value explicitly.

As for potential issues of using pruning, the docs doesn't recommended to use pruning in production but I believe this advice could be updated and for Rococo, I am using `--pruning=1000 --unsafe-pruning` without issues.

**Comment by dblane-digicatapult** [comment]
Tested with

- pruning not set
- `pruning: 1000`
- `pruning: 0`

File: charts/node/templates/statefulset.yaml [hunk 1]
```diff
@@ -265,6 +265,13 @@ spec:
                 {{- if or (eq .Values.node.role "authority") (eq .Values.node.role "validator") }}
                 --validator \
                 {{- end }}
+                {{- if ge ( int .Values.node.pruning ) 1 }}
+                --unsafe-pruning \
+                --pruning={{ .Values.node.pruning }} \
+                {{- else if and ( not ( kindIs "invalid" .Values.node.pruning ) ) ( eq 0 ( int .Values.node.pruning ) ) }}
+                --unsafe-pruning \
```
Comment (PierreBesson) [inline charts/node/templates/statefulset.yaml:272]: `--unsafe-pruning` will not be needed for `pruning=archive`

File: charts/node/templates/statefulset.yaml [hunk 2]
```diff
@@ -265,6 +265,13 @@ spec:
                 {{- if or (eq .Values.node.role "authority") (eq .Values.node.role "validator") }}
                 --validator \
                 {{- end }}
+                {{- if ge ( int .Values.node.pruning ) 1 }}
+                --unsafe-pruning \
+                --pruning={{ .Values.node.pruning }} \
+                {{- else if and ( not ( kindIs "invalid" .Values.node.pruning ) ) ( eq 0 ( int .Values.node.pruning ) ) }}
```
[the remark this answers is not in the document]
Comment (dblane-digicatapult) [reply charts/node/templates/statefulset.yaml:271]: > It's not clear what is the result when `pruning` is undefined. I assume the block is omitted.

That is correct.

**Review by PierreBesson — APPROVED** [review]

**Pull request merged**